signature.go

```diff
@@ -96,3 +97,62 @@ func Signature(input io.Reader, output io.Writer, blockLen, strongLen uint32, si
 
 	return &ret, nil
 }
+
+// ReadSignatureFile reads a signature from the file at path.
+func ReadSignatureFile(path string) (*SignatureType, error) {
```

Comment:
How about exposing two functions? one that takes a `io.Reader` and another one that wraps it and does the `os.Open` call on a path. That gives full flexibility to any library user. Like if we decide to cache signatures at some point :)

Reply:
Yep, good suggestion, thanks! Looks more like proper Go :-)

Just pushed an updated version!